cmd/p2pdcli/p2pdcli.go

```diff
@@ -92,14 +95,24 @@ func main() {
 }
 
 // getConnection returns a client for a GRPC server at the specified address.
-func getConnection(serverAddress *string) (*grpc.ClientConn, error) {
+func getConnection(serverAddress *string, serverCertificate *string) (*grpc.ClientConn, error) {
 	if *serverAddress == "" {
 		return nil, errors.New("No server address provided")
 	}
 
 	var opts []grpc.DialOption
 
-	opts = append(opts, grpc.WithInsecure())
+	if *serverCertificate == "" {
+		opts = append(opts, grpc.WithInsecure())
+		return grpc.Dial(*serverAddress, opts...)
+	}
+
+	creds, err := credentials.NewClientTLSFromFile(*serverCertificate, "")
+
+	if err != nil {
+		panic(err)
```

Comment:
maybe return the error as you are handling the error (add message and log) on return

Reply:
Good point done.